diff --git a/hw/xwayland/xwayland-present.c b/hw/xwayland/xwayland-present.c
index 1a99228a1b475652a72798c14027f8e5c1817588..7ae13e649f0097e71bd145277e50ee5c5fa54ec8 100644
--- a/hw/xwayland/xwayland-present.c
+++ b/hw/xwayland/xwayland-present.c
@@ -98,14 +98,16 @@ xwl_present_has_pending_events(struct xwl_present_window *xwl_present_window)
            !xorg_list_is_empty(&xwl_present_window->wait_list);
 }
 
-static void
+void
 xwl_present_reset_timer(struct xwl_present_window *xwl_present_window)
 {
     if (xwl_present_has_pending_events(xwl_present_window)) {
+        struct xwl_window *xwl_window = xwl_window_from_window(xwl_present_window->window);
         CARD32 now = GetTimeInMillis();
         CARD32 timeout;
 
-        if (!xorg_list_is_empty(&xwl_present_window->frame_callback_list))
+        if (xwl_window && xwl_window->frame_callback &&
+            !xorg_list_is_empty(&xwl_present_window->frame_callback_list))
             timeout = TIMER_LEN_FLIP;
         else
             timeout = TIMER_LEN_COPY;
@@ -359,8 +361,8 @@ xwl_present_queue_vblank(WindowPtr present_window,
 
     xorg_list_append(&event->list, &xwl_present_window->wait_list);
 
-    /* If there's a pending frame callback, use that */
-    if (xwl_window && xwl_window->frame_callback &&
+    /* Hook up to frame callback */
+    if (xwl_window &&
         xorg_list_is_empty(&xwl_present_window->frame_callback_list)) {
         xorg_list_add(&xwl_present_window->frame_callback_list,
                       &xwl_window->frame_callback_list);
@@ -495,9 +497,6 @@ xwl_present_flip(WindowPtr present_window,
     /* We can flip directly to the main surface (full screen window without clips) */
     wl_surface_attach(xwl_window->surface, buffer, 0, 0);
 
-    if (!xwl_window->frame_callback)
-        xwl_window_create_frame_callback(xwl_window);
-
     if (xorg_list_is_empty(&xwl_present_window->frame_callback_list)) {
         xorg_list_add(&xwl_present_window->frame_callback_list,
                       &xwl_window->frame_callback_list);
@@ -506,6 +505,9 @@ xwl_present_flip(WindowPtr present_window,
     /* Realign timer */
     xwl_present_reset_timer(xwl_present_window);
 
+    if (!xwl_window->frame_callback)
+        xwl_window_create_frame_callback(xwl_window);
+
     xwl_surface_damage(xwl_window->xwl_screen, xwl_window->surface,
                        damage_box->x1 - present_window->drawable.x,
                        damage_box->y1 - present_window->drawable.y,
diff --git a/hw/xwayland/xwayland-present.h b/hw/xwayland/xwayland-present.h
index 40bd7945988f11f32d1675a7f3623f4119390887..ce91f557bc6b1b3accde64d470d2efe3e920cb6a 100644
--- a/hw/xwayland/xwayland-present.h
+++ b/hw/xwayland/xwayland-present.h
@@ -64,6 +64,7 @@ struct xwl_present_event {
     struct xorg_list list;
 };
 
+void xwl_present_reset_timer(struct xwl_present_window *xwl_present_window);
 void xwl_present_frame_callback(struct xwl_present_window *xwl_present_window);
 Bool xwl_present_init(ScreenPtr screen);
 void xwl_present_cleanup(WindowPtr window);
diff --git a/hw/xwayland/xwayland-window.c b/hw/xwayland/xwayland-window.c
index 92bcae326e071f6bd0bb81e79cae9b551b033353..41061e3448f038416015c30cb206a5003f2edeeb 100644
--- a/hw/xwayland/xwayland-window.c
+++ b/hw/xwayland/xwayland-window.c
@@ -765,6 +765,18 @@ xwl_window_create_frame_callback(struct xwl_window *xwl_window)
     xwl_window->frame_callback = wl_surface_frame(xwl_window->surface);
     wl_callback_add_listener(xwl_window->frame_callback, &frame_listener,
                              xwl_window);
+
+#ifdef GLAMOR_HAS_GBM
+    if (xwl_window->xwl_screen->present) {
+        struct xwl_present_window *xwl_present_window, *tmp;
+
+        xorg_list_for_each_entry_safe(xwl_present_window, tmp,
+                                      &xwl_window->frame_callback_list,
+                                      frame_callback_list) {
+            xwl_present_reset_timer(xwl_present_window);
+        }
+    }
+#endif
 }
 
 Bool
